src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -94,26 +95,26 @@ public static double findBestAngle(){
     return angles[angleIndex];
   }
 
-  public static void calculateShooter(){
+  public void calculateShooter(){
     angle = findBestAngle();
-    velocity = findVelocity(distance);
 
-    
-    System.out.println("best angle: " + angle);
+    double originalDistance = distance;
+    double xOffset = (elevatorPivotLength*Math.sin(angle - flywheelOffset)) + (shooterFromElevator*Math.sin(Units.degreesToRadians(90) - angle));
+    distance = originalDistance - xOffset + elevatorXOffset;
+    velocity = findVelocity(distance);
 
     angle = Units.degreesToRadians(angle);
-    // height = elevatorPivotHeight-(elevatorPivotLength*Math.cos(angle - flywheelOffset)) + (shooterFromElevator*Math.sin(angle));
-    // double originalDistance = distance;
-    double xOffset = (elevatorPivotLength*Math.sin(angle - flywheelOffset)) + (shooterFromElevator*Math.sin(Units.degreesToRadians(90) - angle));
     height = elevatorPivotHeight-(elevatorPivotLength*Math.cos(angle - flywheelOffset)) + (shooterFromElevator*Math.sin(angle));
-    // distance = originalDistance - xOffset + elevatorXOffset;
-    System.out.println("x offset: " + xOffset);
-    System.out.println("height: " + height);
     pivotAngle = angle - Units.radiansToDegrees(flywheelOffset);
-    System.out.println("pivot angle: " + pivotAngle);
   }
 
-  public static void main(String[] args){
-    calculateShooter();
+  public double getPivotAngle(){
+    return Units.radiansToDegrees(pivotAngle);
+  }
+  public double getRpm(){
```

Comment:
Please set the velocity conversion factor on the motors, don't deal with RPMs in the code

Reply:
The rpmFromMPS method is just to convert the needed meters/sec to rpm. I don't know what the getRpm method is for though. @SpeedyCoder37